### DISCUSSION TITLE
Arduino GIGA R1 WiFi

### DISCUSSION THREAD

**Discussion by Ancyker** [sanni/cartreader#753] [Ideas & Suggestions]
Well, the [Arduino GIGA R1 WiFi](https://docs.arduino.cc/hardware/giga-r1-wifi) is out. It's based around an STM32 (specifically, an [STM32H747XI](https://www.digikey.com/en/products/detail/stmicroelectronics/STM32H747XIH6/10244387)) with 2 32-bit cores, one running at 480MHz and the other at 240MHz. It's got WiFi and Bluetooth, as well as a USB-C port. If that wasn't enough, there's also an integrated display controller with a connector built right into the board. Did I mention the 76 GPIOs? Yeah.

Now for the cons. It's sold out basically everywhere, but it hasn't been out that long. The same thing happened with the Pi Pico and Pico W when they came out, and now they are plentiful, so time will tell. But, one thing time won't help much with is price. It's listed at $70, that's pretty much the same price as the overkill-for-this [STM32 discovery kit](STM32MP157D) I've been tinkering with. Oh, and it's using the MEGA footprint, so it wouldn't work with HW5 (unless we make an adapter?), though it could be made to work with the existing six-slot adapter.

With that said, I did manage to order one, so I'll see if it can be worked in. Due to the stock issues, I had to order it from the UK (I'm in the USA), so it'll be quite a bit before it comes (it's currently in Germany?). Still, figured I'd post for feedback/thoughts/etc.

**Comment by sanni** [comment]
Lots of interesting new micros coming out soon like the [Arduino Uno R4](https://store.arduino.cc/pages/unor4) with a Renesas Arm-RA4M1 chip, ofc it doesn't have enough pins but maybe they make a Arduino Mega R4 too in the future. Also the [ESP32-P4](https://www.espressif.com/en/news/ESP32-P4) seems promising. 

And then there are older chips like the STM32F070RB(and similar) that the [INLretro Dumper-Programmer](https://www.infiniteneslives.com/inlretro.php) uses which is neat too.
